rust/agama-dbus-server/src/main.rs

```diff
@@ -4,15 +4,22 @@ pub mod questions;
 
 use agama_network::NetworkService;
 use std::future::pending;
+use systemd_journal_logger::JournalLog;
+use log::LevelFilter;
 
 const ADDRESS: &str = "unix:path=/run/agama/bus";
 
 #[async_std::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    JournalLog::default().install().unwrap();
```

Comment:
`unwrap`...
should be used when you *know* better than the compiler that it cannot fail

The function is already returning a `Result`

```suggestion
    JournalLog::default().install()?;
```

Reply:
... and it turns out that you are right, it fails only if the logger has been set already
https://docs.rs/log/0.4.17/log/fn.set_boxed_logger.html

In that case I'd add a comment *explaining* why it can't fail. Might as well keep the `?`.

Reply:
ok, will document it. As I am not big fan of Box<Error> due to its wide scope and with unwrap it will make it change to more precise error definition easier.

Reply:
> As I am not big fan of Box due to its wide scope and with unwrap it will make it change to more precise error definition easier.

Just some nitpicking while I wait for the `rustfmt` CI run :smile: 

I think it is the `dyn` part, not the `Box`, that you object to :)

In general yes, it makes the error API too catch-all... but then we are in `fn main` and we can only crash... which also refutes my initial objection :rofl: